src/Microsoft.ML.Data/Transforms/ConversionsExtensionsCatalog.cs

```diff
@@ -47,22 +48,24 @@ public static HashingEstimator Hash(this TransformsCatalog.ConversionTransforms
         /// <param name="outputColumnName">Name of the column resulting from the transformation of <paramref name="inputColumnName"/>.</param>
         /// <param name="inputColumnName">Name of the column to transform. If set to <see langword="null"/>, the value of the <paramref name="outputColumnName"/> will be used as source.</param>
         /// <param name="outputKind">The expected kind of the output column.</param>
+        /// <param name="outputKeyCount">New key count, if we work with key type.</param>
         /// <example>
         /// <format type="text/markdown">
         /// <![CDATA[
         ///  [!code-csharp[ConvertType](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/Conversion/ConvertType.cs)]
         /// ]]></format>
         /// </example>
         public static TypeConvertingEstimator ConvertType(this TransformsCatalog.ConversionTransforms catalog, string outputColumnName, string inputColumnName = null,
-            DataKind outputKind = ConvertDefaults.DefaultOutputKind)
-            => new TypeConvertingEstimator(CatalogUtils.GetEnvironment(catalog), outputColumnName, inputColumnName, outputKind);
+            DataKind outputKind = ConvertDefaults.DefaultOutputKind, KeyCount outputKeyCount = null)
```

Comment:
>KeyCount outputKeyCount = null [](start = 69, length = 30)

We have FYI never found this useful and I'd recommend *not* including this. Even though it represents technically a loss of capability, it was a misguided capability.

Reply:
Sounds good I removed it from the extension method, so it will not longer be exposed. 

---
In reply to: [265993731](https://github.com/dotnet/machinelearning/pull/2959#discussion_r265993731) [](ancestors = 265993731)